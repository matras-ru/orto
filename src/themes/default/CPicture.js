export const base = 'w-full object-cover';

export default {
    base
};
