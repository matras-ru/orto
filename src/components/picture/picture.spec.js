/* eslint-env jest */
import { mount } from '@vue/test-utils';
import CPicture from './picture';

const baseClass = 'w-full object-cover';

describe('CPicture', () => {
    it('rendered', () => {
        const wrapper = mount(CPicture, {
            context: {
                props: {
                    src: 'picture1.jpg'
                }
            }
        });

        const img = wrapper.find('img');

        expect(wrapper.isFunctionalComponent).toBe(true);
        expect(wrapper.element.tagName).toEqual('PICTURE');

        expect(img.classes().sort()).toEqual(`${baseClass}`.split(' ').sort());
        expect(img.attributes('src')).toEqual('picture1.jpg');
    });

    it('creating media source', () => {
        const wrapper = mount(CPicture, {
            context: {
                props: {
                    src: 'picture1.jpg',
                    sm: 'picture2.jpg',
                    lg: 'picture3.jpg'
                }
            }
        });

        const img = wrapper.find('img');
        const sources = wrapper.findAll('source');

        expect(wrapper.isFunctionalComponent).toBe(true);
        expect(wrapper.element.tagName).toEqual('PICTURE');

        expect(img.classes().sort()).toEqual(`${baseClass}`.split(' ').sort());
        expect(img.attributes('src')).toEqual('picture1.jpg');

        expect(sources.at(0).attributes('type')).toEqual('image/jpg');
        expect(sources.at(0).attributes('srcset')).toEqual('picture2.jpg');
        expect(sources.at(0).attributes('media')).toEqual('(min-width: 640px)');

        expect(sources.at(1).attributes('type')).toEqual('image/jpg');
        expect(sources.at(1).attributes('srcset')).toEqual('picture3.jpg');
        expect(sources.at(1).attributes('media')).toEqual('(min-width: 1024px)');
    });
});
