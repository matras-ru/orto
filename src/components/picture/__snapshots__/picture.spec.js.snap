// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CPicture creating media source 1`] = `
<picture>
  <source type="image/jpg" srcset="picture3.jpg" media="(min-width: 1024px)">
  <source type="image/jpg" srcset="picture2.jpg" media="(min-width: 640px)"><img src="picture1.jpg" class="w-full object-cover">
</picture>
`;

exports[`CPicture rendered 1`] = `<picture><img src="picture1.jpg" class="w-full object-cover"></picture>`;
